### ISSUE TITLE
Exciting progress

### ISSUE THREAD

**Issue by GeorgeS2019** [stuarta0/godot-cesium#1]
![msedge_vpO5VpX6Qt](https://github.com/stuarta0/godot-cesium/assets/49812372/5307d7e5-c797-4af4-a10f-b01d29cb9f39)

https://youtu.be/LQ4oa8ZQp0g?list=PLFdndNkL5ZR2D72PR6TgRRIJLbNpCbqTY

https://github.com/Lecrapouille/CiudadSim/issues/1#issue-2229740942

https://github.com/godotengine/godot-proposals/issues/1749#issuecomment-2041684921

https://www.buildingtheopenmetaverse.org/episodes/open-source-game-engines

**Comment by stuarta0** [comment]
Hey @GeorgeS2019 - this looks awesome! Is this a GDExtension built in parallel to what I started? Or did it use this repo as a starting point? Do you have a repo or PR?

**Comment by GeorgeS2019** [comment]
@stuarta0 

No, I assume the youtube is from you?

I forgot the author. He once shared that it will be c++ GDExtension

I hope he soon will share the repos

**Comment by stuarta0** [comment]
Nope, not me I'm afraid. UI looks Chinese and they're very light on details from the YouTube channel so it's hard to find any further information. Their first upload certainly looks like the starting point I made in this repo, but maybe it's just a case of simultaneous invention.

**Comment by GeorgeS2019** [comment]
@stuarta0 
I once was contacted (~> 4 weeks) on one of my Godot Cesium posts which I can not track now. 

He is working on it using c++ GDExtension.

we need to wait. 

I have not check your repos. 

Perhaps you could share where are you so others could contribute know what to do.

**Comment by GeorgeS2019** [comment]
This looks great too :-)

https://github.com/stuarta0/launceston-3d-godot

**Comment by stuarta0** [comment]
I only got as far as bootstrapping the extension so the Cesium native library could be compiled alongside the GDExtension using cmake, with a couple of example nodes (Cesium3DTileset). This would've been a great starting point for someone to jump in as it's all the necessary building blocks to hit the ground running.

Since then I'd started digging through cesium native to see what was provided, and cross referencing with the cesium unreal repo to see how it's used. I just haven't had the time to continue trawling code and incrementing the implementation. But it looks like this other person has made great progress!

**Comment by GeorgeS2019** [comment]
@stuarta0 

Keep it up. Let see if I could persuade him to work together. Let us be positive :-)  Good Job!!

https://github.com/godotengine/godot-proposals/discussions/6913

**Comment by GeorgeS2019** [comment]
@stuarta0 

@luxun0517 is making huge progress in both new Godot and cesium native commits

![image](https://github.com/stuarta0/godot-cesium/assets/49812372/f2f705c1-c8d4-4727-976c-2e7610fb6987)

**Comment by stuarta0** [comment]
@GeorgeS2019 I think they're 99% catch up merges with upstream Godot. The couple of relevant Cesium commits are fairly bare bones. Also, I really hope they aren't developing directly against the Godot codebase - it should be against the godot-cpp or other GDExtension API libraries.

**Comment by GeorgeS2019** [comment]
@stuarta0 
I understand that it is GDExtension. Thx for feedback.

**Comment by luxun0517** [comment]
https://www.youtube.com/watch?v=PFO9Gbq0Njc

@stuarta0 Thanks for Your Repo, that is why i use godot

**Comment by GeorgeS2019** [comment]
@stuarta0 because of your pioneering work on Godot for Cesium, @luxun0517 is able to make progress with all these impressive youtubes

**Comment by GeorgeS2019** [comment]
@stuarta0 
Still no progress in the Cesium plugin for Godot?

**Comment by luxun0517** [comment]
> [@stuarta0](https://github.com/stuarta0) Still no progress in the Cesium plugin for Godot?

[https://x.com/harris_luxun/status/1884412817618657560?s=46&t=oBC8_xtCMN138lcHc5xGbg](url)

**Comment by luxun0517** [comment]
> [@stuarta0](https://github.com/stuarta0) Still no progress in the Cesium plugin for Godot?

[cesium multi platform test](https://x.com/harris_luxun/status/1884412817618657560?s=46&t=oBC8_xtCMN138lcHc5xGbg)

**Comment by GeorgeS2019** [comment]
In a few weeks, the plugin will be released

**Comment by luxun0517** [comment]
Nice!!!

**Comment by GeorgeS2019** [comment]
@stuarta0 

https://github.com/wxzen/godot-3dtiles/issues/2#issuecomment-2820000570

Now archive

**Comment by GeorgeS2019** [comment]
https://github.com/meshula/usd-godot

**Comment by NeonfireStudio** [comment]
How's the project going? Is it came in a stable position, so we can try it?

**Comment by GeorgeS2019** [comment]
U need to know how to compile...I have not tried yet

**Comment by GeorgeS2019** [comment]
https://github.com/godotengine/godot-proposals/discussions/6913#discussioncomment-13066998


https://github.com/Battle-Road-Labs/3D-Tiles-For-Godot

**Comment by GeorgeS2019** [comment]
### Sept 2026 Update

https://github.com/brynnb/cesium-godot/issues/7#issue-5334149586
